### PR TITLE
Fix crossgen2 build

### PR DESCRIPTION
**Pull request by davidwrighton**
- Include osx arm64 jit in package
  - Also use that jit when appropriate
- build for all rids
- build cross targeting test scenario variant from separate project file to avoid issue with incorrect app host rid
  - Remove app host from cross targeting x64 crossgen used in test scenarios

Addresses #50514

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/crossgen-contrib

**Review by trylek — APPROVED** [review]
Very nice, thank you!

File: src/coreclr/tools/aot/crossgen2/crossgen2.props [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    <PackageReference Include="Microsoft.DiaSymReader.Native"
+      Version="$(MicrosoftDiaSymReaderNativeVersion)"
+      IsImplicitlyDefined="true"
+      ExcludeAssets="all"
+      GeneratePathProperty="true"
+      />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(TargetOS)' == 'windows'">
+    <!-- DiaSymReader for the target architecture, which is placed into the package -->
```
Comment (AntonLapounov) [inline src/coreclr/tools/aot/crossgen2/crossgen2.props:112]: Nit: This comment should be removed.

**Review by AntonLapounov — APPROVED** [review]
Thank you!

File: src/coreclr/tools/aot/crossgen2/crossgen2_crossarch.csproj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <CrossHostArch>x64</CrossHostArch>
+    <OutputPath>$(RuntimeBinDir)/$(CrossHostArch)/crossgen2</OutputPath>
+    <UseAppHost>false</UseAppHost>
```
[the remark this answers is not in the document]
Comment (davidwrighton) [reply src/coreclr/tools/aot/crossgen2/crossgen2_crossarch.csproj:5]: The two of them make the logic simpler in any case and avoid weird msbuild special purposing. I'm good with keeping the variation here.

File: src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs [hunk 3]
```diff
@@ -141,6 +141,11 @@ private static string GetTargetSpec(TargetDetails target)
                 _ => throw new NotImplementedException(target.Architecture.ToString())
             };
 
+            if ((target.Architecture == TargetArchitecture.ARM64) && (target.OperatingSystem == TargetOS.OSX))
+            {
+                targetOSComponent = "unix_osx";
```
[the remark this answers is not in the document]
Comment (davidwrighton) [reply src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs:146]: What does this have to do with project file TargetSpecs? They are a different concept. This is just the jit naming rules which have never matched the project file target spec logic. (Unix vs linux, etc.)

[the remark this answers is not in the document]
Comment (AntonLapounov) [reply src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs:146]: The `GetTargetSpec` method: https://github.com/dotnet/runtime/blob/31c28fc379ec6f581c101f630a3a81e0f215c1ea/src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs#L134-L144 mimics `TargetSpec` evaluation in crossgen2.csproj:
https://github.com/dotnet/runtime/blob/31c28fc379ec6f581c101f630a3a81e0f215c1ea/src/coreclr/tools/aot/crossgen2/crossgen2.csproj#L62-L68

[the remark this answers is not in the document]
Comment (davidwrighton) [reply src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs:146]: Oh I see. Thanks. Yes, I need to update the handling in the project file. Its not handling the osx arm64 case correctly. (That explains the build breaks)

**Comment by omajid** [comment]
As far as I can tell, this breaks building runtime on arm64 (like we do on Fedora). On an arm64 machine:

```
  crossgen2 -> /home/tester/runtime/artifacts/bin/coreclr/Linux.arm64.Release/crossgen2/crossgen2.dll
/home/tester/runtime/.dotnet/sdk/6.0.100-preview.2.21155.3/Microsoft.Common.CurrentVersion.targets(4941,5): error MSB3030: Could not copy the file "/home/tester/runtime/artifacts/bin/coreclr/Linux.arm64.Release/x64/libjitinterface_x64.so" because it was not found. [/home/tester/runtime/src/coreclr/tools/aot/crossgen2/crossgen2_crossarch.csproj]
```

File: eng/Subsets.props [hunk 4]
```diff
@@ -204,11 +204,13 @@
                              $(CoreClrProjectRoot)tools\r2rtest\R2RTest.csproj;
                              $(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2.csproj" Category="clr" />
 
+    <ProjectToBuild Condition="'$(TargetArchitecture)' != 'x64'" Include="$(CoreClrProjectRoot)tools\aot\crossgen2\crossgen2_crossarch.csproj" Category="clr" />
```
Comment (omajid) [inline eng/Subsets.props:207]: This assumes that everyone only builds on x64. That breaks all ports to all other platforms that are doing non-cross builds...

Comment (AntonLapounov) [reply eng/Subsets.props:207]: Thank you for reporting this issue. I think the condition here and 6 lines below that line should be `'$(TargetArchitecture)' != 'x64' and '$(BuildArchitecture)' == 'x64'`.  I'll submit a PR after verification.

**Pull request merged**